templates/assets/cloudformation/elb.yml

```diff
@@ -135,7 +135,7 @@ Resources:
               AWS:
                 Fn::Sub:
                   - "arn:${AWS::Partition}:iam::${ElbAccountId}:root"
-                  - ElbAccountId: !FindInMap [Regions, !Ref "AWS::Region", ElbAccountId]
+                  - ElbAccountId: !FindInMap [Regions, Ref: "AWS::Region", ElbAccountId]
```

Comment:
Was this change needed, particularly?

Reply:
> Reviewed. There are some stylistic changes I'd ask you to make. Your code is most likely valid, it's just a difficult read and will be more difficult to maintain. Thanks!

Actually the original syntax gave me a hard time. As the template is parsed and merged with other snippets coming as extensions it turned out to be the only working solution. I'm also not a fan of mixing yaml and json syntax but hose are nested constructs.

Reply:
> Was this change needed, particularly?

`- ElbAccountId: !FindInMap [Regions, !Ref "AWS::Region", ElbAccountId]`

Here is what I get:

`18:29:58.354 aws ▶ 061   Create stack complete err=ValidationError: Template error: Unable to get mapping for Regions::AWS::Region::ElbAccountId`

Reply:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-findinmap.html

> You can't nest two instances of two functions in short form.

```
Fn::FindInMap : [Regions, !Ref "AWS::Region", ElbAccountId]
```

^ would be preferred to use.

I'm honestly surprised that `ElbAccountId: !FindInMap [Regions, Ref: "AWS::Region", ElbAccountId]` might work as well since "Ref:" is not an intrinsic function AFAIK.

Reply:
> https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-findinmap.html
> 
> > You can't nest two instances of two functions in short form.
> 
> ```
> Fn::FindInMap : [Regions, !Ref "AWS::Region", ElbAccountId]
> ```
> ^ would be preferred to use.
> 
> I'm honestly surprised that `ElbAccountId: !FindInMap [Regions, Ref: "AWS::Region", ElbAccountId]` might work as well since "Ref:" is not an intrinsic function AFAIK.

Let me answer with this ;):
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference-ref.html

Ultimately both versions are correct. However the template that finally will be deployed is generated. So you need to count in the yaml parser. I'm not familiar with your codebase but the error I mentioned earlier in this conversation indicates that the rendered result is not functioning with `!Ref`.

Reply:
How about merging the PR and releasing 1.5.11 :) ?

Reply:
Resolving this part of the PR review, as there is a good reason (YAML syntax limitation) to use "Ref:" instead of "!Ref" in the array.

Leaving this PR open so Michael can investigate whether the later conditionals have to be in that form or whether any problems using the more verbose form can be resolved.

Reply:
ugh, I had a typo in my original comment. I meant to type out the long form of Ref. Anyway...

Thanks for the link to the docs on Ref. I've corrected that several times as error not realizing it's valid without the preceding !